spec/jobs/cron_job_spec.rb

```diff
@@ -0,0 +1,106 @@
+require "rails_helper"
+
+RSpec.describe "CronJob" do
+  def queue_adapter_for_test
+    DelayedJobTestAdapter.new
+  end
+
+  describe ".schedule" do
+    it "schedules the job" do
+      expect { TestCronJob.schedule }.to change { TestCronJob.send(:jobs).count }.by(1)
+    end
+
+    it "schedules the job with the cron expression" do
+      TestCronJob.schedule
+
+      expect(TestCronJob.send(:scheduled_job).cron).to eq(TestCronJob.cron_expression)
+    end
+
+    context "when the job is not configured to perform on schedule and an unscheduled run of the job exists" do
+      before :each do
+        TestCronJob.perform_later
+      end
+
+      it "leaves the job enqueued" do
+        existing_job = TestCronJob.send(:enqueued_job)
+
+        TestCronJob.schedule
+
+        expect(TestCronJob.send(:enqueued_job)).to eq(existing_job)
+      end
+
+      it "doesn't enqueue another instance of the job" do
+        expect { TestCronJob.schedule }.not_to(change { TestCronJob.send(:jobs).where(cron: nil).count })
```

Comment:
I think it might be clearer to just check the count here, so:

```ruby
schedule_count = TestCronJob.send(:jobs).where(cron: nil).count

TestCronJob.schedule

expect(TestCronJob.send(:jobs).where(cron: nil).count).to eq(schedule_count)
```